addons/product/tests/test_variants.py

```diff
@@ -359,3 +359,21 @@ def test_update_variant_with_nocreate(self):
             variant_id.attribute_value_ids += self.size_S
         template.attribute_line_ids += template.attribute_line_ids.browse()
         self.assertEqual(len(template.product_variant_ids), 1)
+
+
+class TestVariantBarcodes(common.TestProductCommon):
+
+    def setUp(self):
+        res = super(TestVariantBarcodes, self).setUp()
+        self.color_attr = self.env['product.attribute'].create({'name': 'Size'})
+        self.color_attr_value_b = self.env['product.attribute.value'].create({'name': 'Black', 'attribute_id': self.color_attr.id})
+        self.color_attr_value_r = self.env['product.attribute.value'].create({'name': 'Red', 'attribute_id': self.color_attr.id})
+        self.color_attr_value_g = self.env['product.attribute.value'].create({'name': 'Green', 'attribute_id': self.color_attr.id})
```

Comment:
self.env['product.attribute.value'] should better be taken into a variable and then reused?